File: backend/src/main/java/es/udc/paproject/backend/model/services/LesionService.java
package es.udc.paproject.backend.model.services;

import java.util.List;

import es.udc.paproject.backend.model.entities.Lesion;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.UsedLesionException;

public interface LesionService {
    
    Lesion addLesion(String lesionName, String description, String medication, String lesionType) throws InstanceNotFoundException;

    void addLesionToPlayer(Long playerId, Long lesionId) throws InstanceNotFoundException;

    Lesion findLesionById(Long lesionId) throws InstanceNotFoundException;

    Block<Lesion> findAllLesion(int page, int size) throws InstanceNotFoundException;

    List<Lesion> findAllLesion() throws InstanceNotFoundException;

    Block<Lesion> findLesionByType(String lesionType, int page, int size) throws InstanceNotFoundException;

    List<Lesion> findLesionByType(String lesionType) throws InstanceNotFoundException;

    List<Lesion> findLesionByPlayer(Long playerId) throws InstanceNotFoundException;

    void removeLesion(Long lesionId) throws InstanceNotFoundException, UsedLesionException;

    void removeLesionToPlayer(Long playerId, Long lesionId) throws InstanceNotFoundException;

    Lesion updateLesion(Long lesionId, String lesionName, String description, String medication, String lesionType) throws InstanceNotFoundException;
}

File: frontend/src/backend/lesionService.js
import {config, appFetch} from './appFetch';

export const findLesionById = (lesionId, onSuccess, onErrors) => 
    appFetch(`/lesion/${lesionId}`, config('GET'), onSuccess, onErrors);

export const findAllLesionPage = ({page}, onSuccess, onErrors) => 
    appFetch(`/lesion/page?page=${page}`, config('GET'), onSuccess, onErrors);

export const findAllLesion = (onSuccess, onErrors) => 
    appFetch(`/lesion`, config('GET'), onSuccess, onErrors);

export const findLesionByTypePage = ({lesionType, page}, onSuccess, onErrors) =>
    appFetch(`/lesion/${lesionType}/typeLesion/page?page=${page}`, config('GET'), onSuccess, onErrors);

export const findLesionByType = (lesionType, onSuccess, onErrors) =>
    appFetch(`/lesion/${lesionType}/typeLesion`, config('GET'), onSuccess, onErrors);

export const findLesionByPlayer = (playerId, onSuccess, onErrors) => 
    appFetch(`/lesion/${playerId}/player`, config('GET'), onSuccess, onErrors);

export const addLesion = (lesionName, description, medication, lesionType, onSuccess, onErrors) => 
    appFetch(`/lesion?lesionName=${lesionName}&description=${description}&medication=${medication}&lesionType=${lesionType}`, config('POST'), onSuccess, onErrors);

export const addLesionToPlayer = (playerId, lesionId, onSuccess, onErrors) =>
    appFetch(`/lesion/${playerId}/addLesionToPlayer?lesionId=${lesionId}`, config('POST'), onSuccess, onErrors);

export const updateLesion = (lesionId, lesionName, description, medication, lesionType, onSuccess, onErrors) =>
    appFetch(`/lesion/${lesionId}?lesionName=${lesionName}&description=${description}&medication=${medication}&lesionType=${lesionType}`, config('PUT'), onSuccess, onErrors);

export const removeLesion = (lesionId, onSuccess, onErrors) => 
    appFetch(`/lesion/${lesionId}`, config('DELETE'), onSuccess, onErrors);

export const removeLesionToPlayer = (playerId, lesionId, onSuccess, onErrors) =>
    appFetch(`/lesion/${playerId}/player?lesionId=${lesionId}`, config('DELETE'), onSuccess, onErrors);